### PR TITLE
Move GetData as GetDataInternal, and have GetData as wrapper

### PR DESCRIPTION
**Pull request by carlopi**
Alternative implementation to https://github.com/duckdb/duckdb/pull/19777

Goal here is having a wrapper around specific implementations of `GetData` (now `GetDataInternal`) so that extra checks and testing can be performed there.

As an example, adding test code that at randomly returns `HAVE_MORE_OUTPUT` + empty chunk improve testing coverage and stress testing other parts of the code.
Note that different combinations of operators or new optimizations might still reach these results, since all caller of `GetData` are asked to make no assumptions on a restricted API.

The simple testing code already uncovers some problems, to be fixed independently.

Biggest impact of this PR is on extension development, since if you implement a PhysicalOperator you'll need to change the signature to `GetDataInternal`, but advantage (compared to alternative implementation strategies) is keeping code duckdb/duckdb code cleaner / more logical.

EDIT: [the commit adding testing code](https://github.com/duckdb/duckdb/pull/19791/commits/52ba1c55acde17621aeb1469701fb383fe329bca) has been removed, I don't really have a proper way to feature select for skip the test, I will just have the test fixed and then adding this back

### CODE REVIEW

**Comment by carlopi** [comment]
Failures looks to me unconnected, maybe it could be restarted?

File: src/include/duckdb/execution/physical_operator.hpp [hunk 1]
```diff
@@ -123,7 +123,10 @@ class PhysicalOperator {
 	virtual unique_ptr<LocalSourceState> GetLocalSourceState(ExecutionContext &context,
 	                                                         GlobalSourceState &gstate) const;
 	virtual unique_ptr<GlobalSourceState> GetGlobalSourceState(ClientContext &context) const;
-	virtual SourceResultType GetData(ExecutionContext &context, DataChunk &chunk, OperatorSourceInput &input) const;
+
+	virtual SourceResultType GetDataInternal(ExecutionContext &context, DataChunk &chunk,
```
Comment (pdet) [inline src/include/duckdb/execution/physical_operator.hpp:128]: Should it be protected?

Comment (carlopi) [reply src/include/duckdb/execution/physical_operator.hpp:128]: You are correct, I'll fix this up.

**Comment by pdet** [comment]
Thanks, LGTM, I'll just wait to see if @Tishj has any opinions!

**Review by Tishj — APPROVED** [review]
Thanks, LGTM
I don't immediately see what this will be used for, but it's clear that it can be used to execute some code on every GetData call generically

Perhaps even aid in profiling 🤔

**Comment by carlopi** [comment]
Thank you both.

I think most immediate is adding back: https://github.com/duckdb/duckdb/pull/19791/commits/52ba1c55acde17621aeb1469701fb383fe329bca (that can't be merged since that will uncover a couple of bugs...)

Then other more systematic testing, or indeed profiling/logging/validation.
